src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/JurisdictionToUserMapping.java

```diff
@@ -1,20 +1,39 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam;
 
+import com.netflix.util.Pair;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.stereotype.Component;
 
+import java.util.HashMap;
 import java.util.Map;
 
 @Component
+@EnableConfigurationProperties
 @ConfigurationProperties(prefix = "idam")
 public class JurisdictionToUserMapping {
+
     private Map<String, Credential> users;
 
-    public Map<String, Credential> getUsers() {
-        return users;
+    public void setUsers(Map<String, Map<String, String>> users) {
+        this.users = users
+            .entrySet()
+            .stream()
+            .map(this::createPair)
+            .collect(HashMap::new, (map, pair) -> map.put(pair.first(), pair.second()), HashMap::putAll);
```

Comment:
i could swear yesterday this stream can be simplified to human eyes and not lose the optimisation. only this morning idea popped up after looking second time:

```java
Map<K, V> someMap;

someMap
    .entrySet()
    .stream()
    .map(this::createCredentials)
    // can apply filter for null creds here
    .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));

Map.Entry<K, V> createCredentials(Map.Entry<K, V> entry) {
    Credential cred = new Credential(entry.getValue().get("username"), entry.getValue().get("password"));

    entry.setValue(cred);

    return entry;
}
```

Reply:
Create `AbstractMap.SimpleEntry` then 😉